### DISCUSSION TITLE
insert tags into YAML Front Matter from all obsidian tags or perhaps a list of tags?

### DISCUSSION THREAD

**Discussion by zeltak** [SilentVoid13/Templater#756] [Help]
Hi all

im trying to build a templater template for new files i create and wondering how does one choose or quick add tags when using  YAML Front Matter.

that is i want to use something like this with the `tp.system.suggester` but as opposed to giving it pre defined options like in the example below for `ratings` i want to get all current tags available in obsidian. if that's not available im also ok with piping a text file for it to read the tags from. does that make sense? is that even possible?

example
```
title:  <%* tR += title %>
tags: [READ FROM INTERNAL/FILE? ]
rating:   <% tp.system.suggester(["1🌟-really bad", "2🌟-meh", "3🌟-decent", "4🌟-great","5🌟-amazing!!!!"], ["1", "2", "3", "4","5"], "1" )%>
```

thx so much 

Z

**Comment by AdamBullen** [comment]
I would like a similar function also. The ability to append a tag to the existing list of tags in the frontmatter. 

The function is I am creating a recipe, when I add an ingredient I also want the name of the ingredient added to the frontmatter.

**Comment by shabegom** [comment]
This snippet puts all tags in a suggester. You could use parts of it to do what you're looking for:

```
<%*
const tags = Object.keys(app.metadataCache.getTags())
let selection = await tp.system.suggester([...tags, "Make New"], [...tags, "Make New"])
if (selection === "Make New") {
    selection = await tp.system.prompt("new tag")
}
tR += selection
%>
```

**Comment by shabegom** [comment]
@AdamBullen check out the MetaEdit plugin. It might be the easier way to edit frontmatter tags.
